measure/dependent_by_index.go

```diff
@@ -0,0 +1,60 @@
+package measure
+
+import (
+	"encoding/json"
+)
+
+// DependentIndexed is a measure uses a custom cost func to calculate parameter
+// dependent costs for connecting to points by index.
+func DependentIndexed(
+	timeDependent bool,
+	cost func(
+		from,
+		to int,
+		data VehicleData,
```

Comment:
Ultimately we would have to benchmark it, but I feel like passing a pointer to that struct instead of a copy may be better here. The struct is about 100 bytes big.

Reply:
changed it.